exporter/datadogexporter/internal/hostmetadata/metadata.go

```diff
@@ -150,21 +164,25 @@ func Pusher(ctx context.Context, params exporter.CreateSettings, pcfg PusherConf
 	// All fields that are being filled in by our exporter
 	// do not change over time. If this ever changes `hostMetadata`
 	// *must* be deep copied before calling `fillHostMetadata`.
-	hostMetadata := &payload.HostMetadata{Meta: &payload.Meta{}, Tags: &payload.HostTags{}}
+	hostMetadata := payload.HostMetadata{Meta: &payload.Meta{}, Tags: &payload.HostTags{}}
 	if pcfg.UseResourceMetadata {
 		hostMetadata = metadataFromAttributes(attrs)
 	}
-	fillHostMetadata(params, pcfg, p, hostMetadata)
+	fillHostMetadata(params, pcfg, p, &hostMetadata)
 
 	// Run one first time at startup
-	pushMetadataWithRetry(retrier, params, pcfg, hostMetadata)
+	pusher.Push(ctx, hostMetadata)
```

Comment:
Do you want to check the return error here?

Reply:
Good point, checking it now